src/Pharmacist.Core/Generation/ReflectionExtensions.cs

```diff
@@ -165,14 +165,26 @@ public static string GenerateFullGenericName(this IType currentType)
 
             if (currentType.TypeParameterCount > 0)
             {
-                sb.Append('<')
-                    .Append(string.Join(", ", currentType.TypeArguments.Select(GenerateFullGenericName)))
-                    .Append('>');
+                sb.Append('<');
+                sb.Append(currentType.IsUnbound()
+                    ? string.Join(", ", currentType.TypeArguments.Select(param => param.FullName))
+                    : string.Join(", ", currentType.TypeArguments.Select(GenerateFullGenericName)));
```

Comment:
Try setting a Func before the Append, so it becomes just like 

`currentType.TypeArguments.Select(func)`

Reply:
Done!